PYIC-6069: Update orch stub for MFA reset result

MikeCollingwood opened the pull request:
## Proposed changes

### What changed

- Add mfa-reset-result.mustache
- Extract errors into error.mustache
- Conditionally render on IpvHandler callback result

### Why did it change

- Need to conditionally render mfa reset result based on state value passed in and retrieved by orch stub
- Errors were extracted since no longer tied to a single client.

### Issue tracking

- [PYIC-6069](https://govukverify.atlassian.net/browse/PYIC-6069)

### Screenshots

1. <img width="789" alt="Screenshot 2024-05-24 at 10 42 49" src="https://github.com/govuk-one-login/ipv-stubs/assets/144116535/c579b3a1-96d7-4396-bf53-6d4844ec47d8">


2. <img width="507" alt="Screenshot 2024-05-24 at 10 02 31" src="https://github.com/govuk-one-login/ipv-stubs/assets/144116535/a2f90801-8c3d-4969-8cb7-3e85037a5fc7">


3. The user details page looks unchanged

[PYIC-6069]: https://govukverify.atlassian.net/browse/PYIC-6069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MikeCollingwood committed:
PYIC-6069:
- Add mfa-reset-result.mustache
- Extract errors into error.mustache
- Conditionally render on IpvHandler callback result

MikeCollingwood committed:
PYIC-6069:
- Expand error catching

MikeCollingwood committed:
PYIC-6069:
- Fix to get state from request query map

Wynndow commented on di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/handlers/IpvHandler.java:
```diff
@@ -215,8 +215,7 @@ private URI getIpvBackchannelEndpoint(String environment) throws URISyntaxExcept
             (Request request, Response response) -> {
-                List<Map<String, Object>> mustacheData;
-                Map<String, Object> moustacheDataModel = new HashMap<>();
+                var gson = new GsonBuilder().setPrettyPrinting().create();
```
In core we've recently removed Gson completely in favour of standardising on Jackson. Perhaps we should do the same thing here. You can make it's output pretty(er) with `new ObjectMapper().enable(INDENT_OUTPUT)`
Either way it would be good to just create the object once, rather than for every callback. Maybe set it as a static final field?

Wynndow commented on di-ipv-orchestrator-stub/src/main/resources/templates/error.mustache:
```diff
@@ -0,0 +1,88 @@
+        {{#error}}
+            <h3 class="govuk-heading-l">Something went wrong: </h3>
+            <div class="govuk-details__text">
+                <pre><code>error: {{error}}</code></pre>
```
Should this be `{{error_code}}`?

MikeCollingwood replied:
It hasn't been in the past so was wondering about this. I will do now that you agree with my initial reaction to it!

Wynndow replied:
I acutally don't know if it's correct - if it works, great! I'm not a mustache expert

Wynndow commented on di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/handlers/IpvHandler.java:
```diff
@@ -225,29 +224,37 @@ private URI getIpvBackchannelEndpoint(String environment) throws URISyntaxExcept
+                                "user-info.mustache");
+                    } else if (AUTH_STUB_STATE.toString().equals(state)) {
+                        return ViewHelper.render(
+                                Map.of("data", userInfo), "mfa-reset-result.mustache");
```
Should we be stringifying the `userInfo` blob here too?

Wynndow replied:
Everyone loves pretty

MikeCollingwood replied:
I need to access the success boolean of it so I might have to make there be multiple params to the template

MikeCollingwood committed:
PYIC-6069:
- Display error_code instead of error
- Use object mapper to map auth data
- Add raw object parameter

MikeCollingwood committed:
PYIC-6069:
- Move object mapper definition

sam803 reviewed (approved):
lgtm
